plugin/src/main/cpp/editor/xr_project_setup_dialog.cpp

```diff
@@ -0,0 +1,625 @@
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "editor/xr_project_setup_dialog.h"
+
+#include "editor_plugin.h"
+#include "raw_headers/mr_startup.tscn.gen.h"
+#include "raw_headers/start_mr.gd.gen.h"
+#include "raw_headers/start_vr.gd.gen.h"
+#include "raw_headers/vr_startup.tscn.gen.h"
+
+#include <godot_cpp/classes/button.hpp>
+#include <godot_cpp/classes/center_container.hpp>
+#include <godot_cpp/classes/config_file.hpp>
+#include <godot_cpp/classes/control.hpp>
+#include <godot_cpp/classes/dir_access.hpp>
+#include <godot_cpp/classes/editor_file_system.hpp>
+#include <godot_cpp/classes/editor_interface.hpp>
+#include <godot_cpp/classes/editor_settings.hpp>
+#include <godot_cpp/classes/file_access.hpp>
+#include <godot_cpp/classes/h_box_container.hpp>
+#include <godot_cpp/classes/h_separator.hpp>
+#include <godot_cpp/classes/http_request.hpp>
+#include <godot_cpp/classes/label.hpp>
+#include <godot_cpp/classes/line_edit.hpp>
+#include <godot_cpp/classes/margin_container.hpp>
+#include <godot_cpp/classes/option_button.hpp>
+#include <godot_cpp/classes/os.hpp>
+#include <godot_cpp/classes/panel_container.hpp>
+#include <godot_cpp/classes/popup_menu.hpp>
+#include <godot_cpp/classes/project_settings.hpp>
+#include <godot_cpp/classes/scene_tree.hpp>
+#include <godot_cpp/classes/scroll_container.hpp>
+#include <godot_cpp/classes/texture_rect.hpp>
+#include <godot_cpp/classes/theme.hpp>
+#include <godot_cpp/classes/v_box_container.hpp>
+#include <godot_cpp/classes/window.hpp>
+#include <godot_cpp/core/error_macros.hpp>
+#include <godot_cpp/core/memory.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+static const char *XR_STARTUP_SCENE_PATH = "res://xr_startup.tscn";
+static const char *START_XR_SCRIPT_PATH = "res://start_xr.gd";
+
+enum AlertType {
+	ALERT_TYPE_ERROR,
+	ALERT_TYPE_WARNING,
+	ALERT_TYPE_NONE,
+};
+
+enum ProjectType {
+	PROJECT_TYPE_ANY,
+	PROJECT_TYPE_VR,
+	PROJECT_TYPE_MR,
+};
+
+enum VendorType {
+	VENDOR_TYPE_ANY,
+	VENDOR_TYPE_META,
+	VENDOR_TYPE_HTC,
+	VENDOR_TYPE_PICO,
+	VENDOR_TYPE_LYNX,
+	VENDOR_TYPE_MAGIC_LEAP,
+};
+
+namespace godot_openxr_vendors_internal {
```

Comment:
Is there a reason for using `godot_openxr_vendors_internal` as the namespace? 
Wondering if `godot_openxr_vendors` should be sufficient?